data_upload/extract.py

```diff
@@ -0,0 +1,120 @@
+import os
+import csv
+from datetime import datetime
+from dotenv import load_dotenv
+import boto3
+import psycopg2
+import psycopg2.extras
+from psycopg2.extensions import connection, cursor
+from typing import Any
+
+
+QUERY = """ SELECT
+    e.earthquake_id,
+    e.time,
+    e.magnitude,
+    e.tsunami,
+    e.felt_report_count,
+    e.cdi,
+    e.latitude,
+    e.longitude,
+    e.depth,
+    e.detail_url,
+    a.alert_type,
+    m.magnitude_type,
+    n.network_name,
+    t.type_name
+    FROM earthquakes AS e
+    JOIN alerts AS a ON e.alert_id = a.alert_id
+    JOIN networks AS n ON e.network_id = n.network_id
+    JOIN magnitude AS m ON e.magnitude_id = m.magnitude_id
+    JOIN type AS t ON e.type_id = t.type_id;
+"""
+
+CSV_FILE = f"/tmp/{datetime.today().strftime('%Y-%m-%d')}_data.csv"
+
+
+def get_connection() -> connection:
+    """Gets connection to database"""
+    return psycopg2.connect(
+        user=os.getenv("DB_USER"),
+        password=os.getenv("DB_PASSWORD"),
+        host=os.getenv("DB_HOST"),
+        port=os.getenv("DB_PORT"),
+        database=os.getenv("DB_NAME")
+    )
+
+
+def get_cursor(conn: connection) -> cursor:
+    """Returns cursor for the connection"""
+    return conn.cursor(cursor_factory=psycopg2.extras.RealDictCursor)
+
+
+def extract_data() -> None:
+    """Extracts data from database and writes to a CSV file"""
+    conn = None
+    try:
+        conn = get_connection()
+        cur = get_cursor(conn)
+        print("Executing query...")
+        cur.execute(QUERY)
+        results = cur.fetchall()
+        column_titles = [desc[0] for desc in cur.description]
+
+        with open(CSV_FILE, mode="w", newline="", encoding="utf-8") as csv_file:
+            writer = csv.writer(csv_file)
+            writer.writerow(column_titles)
+            writer.writerows([list(row.values()) for row in results])
```

Comment:
Could pandas be quicker for this?
```suggestion
    df = pd.dataframe(results,columns=column_titles)
    df.to_csv(CSV_FILE)
```

Reply:
Or you could even do `pd.read_sql(QUERY,connection)` rather than doing `curs.execute` to get the table straight into a dataframe with the column titles.